onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h

```diff
@@ -90,14 +101,26 @@ TreeEnsembleCommon<ITYPE, OTYPE>::TreeEnsembleCommon(int parallel_tree, int para
   ORT_ENFORCE(nodes_falsenodeids.size() == nodes_nodeids.size());
   ORT_ENFORCE(nodes_falsenodeids.size() == nodes_treeids.size());
   ORT_ENFORCE(nodes_falsenodeids.size() == nodes_truenodeids.size());
-  ORT_ENFORCE(nodes_falsenodeids.size() == nodes_values.size());
+  ORT_ENFORCE(nodes_falsenodeids.size() == nodes_values.size() || nodes_falsenodeids.size() == nodes_values_as_tensor.size());
   ORT_ENFORCE(target_class_ids.size() == target_class_nodeids.size());
   ORT_ENFORCE(target_class_ids.size() == target_class_treeids.size());
   ORT_ENFORCE(target_class_ids.size() == target_class_treeids.size());
+  ORT_ENFORCE(base_values.size() == 0 || base_values_as_tensor.size() == 0);
+  ORT_ENFORCE(nodes_hitrates.size() == 0 || nodes_hitrates_as_tensor.size() == 0);
+  ORT_ENFORCE(nodes_values.size() == 0 || nodes_values_as_tensor.size() == 0);
+  ORT_ENFORCE(target_class_weights.size() == 0 || target_class_weights_as_tensor.size() == 0);
 
   aggregate_function_ = MakeAggregateFunction(aggregate_function);
   post_transform_ = MakeTransform(post_transform);
-  base_values_ = base_values;
+  if (base_values_as_tensor.size() > 0) {
+    ORT_ENFORCE(base_values.size() == 0);
```

Comment:
!empty() and empty() looks more readable

Reply:
done